perf(transformer): pre-calculate unsupported flags in RegExp transform

overlookmotel opened the pull request:
Calculate unsupported RegExp flags at start, to make checking flags on each RegExp during traversal cheaper (a single OR operation). i.e. do work once at the start, rather than over and over each time you find a `RegExp`.

overlookmotel commented:
* **#5482** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5482?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
That's a much bigger perf boost than I expected...

<img width="776" alt="bench" src="https://github.com/user-attachments/assets/8a9d4356-2b75-4437-87da-f32a4f96d3a6">

Dunqing reviewed (approved):
Well done! Always learn something from your PR 😲

overlookmotel committed:
perf(transformer): pre-calculate unsupported flags in RegExp transform (#5482)

Calculate unsupported RegExp flags at start, to make checking flags on each RegExp during traversal cheaper (a single OR operation). i.e. do work once at the start, rather than over and over each time you find a `RegExp`.
